bishnu-suyel/Druid-Final-Project
# Tdfp 44

## In this PR
1. Extracted the `contact_id` from the Mautic cookie.
2. Passed the `contact_id` to Drupal’s endpoint.
3. Created custom mautic_integration module: 
- creates an endpoint to receive contact_id from React;
- Queries the Mautic API to get the visitor’s segments by contact_id;
- Checks if taxonomy terms corresponding to Mautic segments exist in Drupal. If not, creates them;
- sets the value of paragraph's field_visibility to visible/invisible
4. Fetched and filtered paragraphs from JSON API based on the value of the field_visibility and rendered only visible paragraphs.

## To test
1. Import configurations
2. Enable mautic_integration module
3. Connect to Mautic using your credentials.
4. Import Mautic database.
5. Add yourself manually to a segment in Mautic.
6. You should be able to see the paragraph that is selected to match Mautic segment.
